src/genophenocorr/analysis/_commie.py

```diff
@@ -145,12 +146,28 @@ def compare_by_protein_feature_type(self, feature_type: FeatureType, tx_id: str)
         return self._apply_boolean_predicate(predicate)
 
     def compare_by_protein_feature(self, feature: str, tx_id: str) -> GenotypePhenotypeAnalysisResult:
-        # TODO - Lauren implement
-        raise NotImplementedError()
+        predicate = ProtFeaturePredicate(tx_id, feature)
+        return self._apply_boolean_predicate(predicate)
+
+    def compare_by_variant_effects(self, effect1: VariantEffect, effect2: VariantEffect, tx_id: str) -> GenotypePhenotypeAnalysisResult:
+        predicate = VariantEffectsPredicate(tx_id, effect1, effect2)
+        return self._apply_boolean_predicate(predicate)
```

Comment:
Sorry, I missed this one.

`VariantEffectsPredicate` is not a `BooleanPredicate`, right? So, you must not call `_apply_boolean_predicate`:

![image](https://github.com/monarch-initiative/genophenocorr/assets/12170955/548f1337-5d01-4983-9135-2c7aa49db681)

Ideally, your IDE should help you with these bugs. If it doesn't you must check the types at the function boundary. Otherwise, this is a recipe for bugs. :/

---

This can be fixed by calling `_apply_poly_predicate` because `GroupingPredicate` is a `PolyPredicate`.